client/src/templates/Challenges/components/completion-modal.tsx

```diff
@@ -92,6 +92,22 @@ export function getCompletedPercent(
   return completedPercent > 100 ? 100 : completedPercent;
 }
 
+export function getCompletedProjects(
+  completedChallengesIds: string[] = [],
+  currentBlockIds: string[] = [],
+  currentChallengeId: string
+): string {
+  completedChallengesIds = completedChallengesIds.includes(currentChallengeId)
```

Comment:
Adding the `currentChallengeId` here steps away from the central state management. We should not confuse ourselves by calling `currentChallengeId` as part of `compeltedChallengesIds`. Instead, including the comment below, this logic should be simplified to count the number of completed **challenges** in the block, and add one if necessary.

Reply:
It's been a while, Do you mind adding code change suggestions or pushing a commit to keep this moving?